loopchain/blockchain/blockchain.py

```diff
@@ -1101,11 +1117,19 @@ def genesis_invoke(self, block: Block, prev_block_ = None) -> ('Block', dict):
         self.__invoke_results[new_block.header.hash] = (tx_receipts, None)
         return new_block, tx_receipts
 
+    def _convert_reps_change_reason(self, state: str) -> NextRepsChangeReason:
+        if state == "0x0":
+            return NextRepsChangeReason.TermEnd
+        elif state == "0x1":
+            return NextRepsChangeReason.Penalty
+
+        return NextRepsChangeReason.NoChange
+
```

Comment:
Is there any reason that this method belongs to `BlockChain` class?

Reply:
Nope. It have to move to proper location.